### ISSUE TITLE
runtime metric: `process.runtime.jvm.gc.duration` publishing odd values

### ISSUE THREAD

**Issue by yorikya** [open-telemetry/opentelemetry-java-instrumentation#8305]
**Describe the bug**
after upgrading from io.opentelemetry.instrumentation/opentelemetry-runtime-metrics "1.16.0-alpha" --> io.opentelemetry.instrumentation/opentelemetry-runtime-metrics "1.24.0-alpha" I had a few issues with this metric:

I don't get the gc:ps_marksweep tag anymore
The numbers I am getting look strange, before the upgrade I had: avg:runtime.jvm.gc.time: 5.39K (gc:ps_scavenge) and after the upgrade I am getting process.runtime.jvm.gc.duration: 12 (gc:ps_scavenge). I know that the metric type has been changed from the gauge to the histogram but the numbers look odd to me
![Screenshot 2023-04-16 at 15 36 03](https://user-images.githubusercontent.com/46996631/232314013-f1385d31-f71a-4445-864f-9ae2ae2dc5d5.png)

related to this [MR](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/6964#issuecomment-1510361877) 

**Steps to reproduce**
Upgrade opentelemetry-runtime-metrics from "1.16.0-alpha" to "1.24.0-alpha"

**What did you expect to see?**
I am expecting to see the same values from the old and the new metric

**What did you see instead?**
I see odd values for the GC duration

**What version are you using?**
1.24.0-alpha

**Environment**
Compiler: java8apt
OS: Ubuntu 14.04.6 LTS 
Runtime: java version "1.8.0_181"
Java(TM) SE Runtime Environment (build 1.8.0_181-b13)
Java HotSpot(TM) 64-Bit Server VM (build 25.181-b13, mixed mode)

**Comment by laurit** [comment]
Although the title of the second chart says `Time By GC Total` looking at the line that goes up and down it doesn't seem to be what is shown on the chart. I think that this chart is probably showing the time of the individual gcs or the average time they took instead of the cumulative sum you want. Consult the documentation of your metrics backend on how you could apply an analytic function such as `sum` on the chart.

**Comment by yorikya** [comment]
Thanks @laurit , I have applied the cumulative sum on the metric now I am monitoring the values.

Do you know why I don't get the `gc:ps_marksweep` tag anymore within the new metric?

**Comment by laurit** [comment]
@yorikya what values do you get for `gc` and `action` attributes?

**Comment by yorikya** [comment]
Now with the `cumsum` looks like I am getting the right values (for the `gc:ps_scavenge`), but I am still missing values for the `gc:ps_marksweep` tag

**Comment by PeterF778** [comment]
@yorikya Please double check your setup. I'm seeing both _scavenge_ and _marksweep_ GC's in Prometheus, using 1.24 agent.
<img width="1427" alt="Screenshot 2023-04-17 at 5 31 29 PM" src="https://user-images.githubusercontent.com/91101705/232639324-53040133-d6d3-400b-a155-d8c279af6356.png">

**Comment by yorikya** [comment]
OK, I will check again 👍

**Comment by yorikya** [comment]
I have confirmed I don't see the `marksweep`, I have tried via logging exporter but I don't see the tag
Also, I have a question @PeterF778 why do you check the `process.runtime.jvm.gc.duration.sum` , with the `sum` suffix? 
I am asking because we are using the Datadog as the backend for the metrics they add `sum/count` to each Histogram (Datadog: Distribution) metric automatically 

The only output from the Logging:
```
StartTimestamp: 2023-04-17 09:29:17.947 +0000 UTC
Timestamp: 2023-04-18 13:17:27.947 +0000 UTC
Value: 28
Metric #7
Descriptor:
     -> Name: process.runtime.jvm.gc.duration
     -> Description: Duration of JVM garbage collection actions
     -> Unit: ms
     -> DataType: Histogram
     -> AggregationTemporality: Delta
HistogramDataPoints #0
Data point attributes:
     -> action: Str(end of minor GC)
     -> gc: Str(PS Scavenge)
```

**Comment by PeterF778** [comment]
Oh, this is just how Prometheus aggregates the histogram data. I can also view gc count and the original histogram, but I find the sum (which corresponds to the old metrics, pre-1.20 release) very useful.

BTW, did you by any chance specify java command line option `-XX:+DisableExplicitGC`? This could be a potential reason for missing the marksweep data.

**Comment by trask** [comment]
@yorikya can you test using the logging metric exporter in order to rule out the backend as a source of issue?

**Comment by yorikya** [comment]
@PeterF778 No, I don't use the `-XX:+DisableExplicitGC` option, I will try today to report the metric direct from the SDK to the STDOUT export (not via the collector), and will check if I see the missing tag, maybe for some reason collector removing this metric 🤷

**Comment by yorikya** [comment]
I have confirmed also in the SDK STDOUT exporter I see only the `gc="PS Scavenge"` tag

```
INFO: metric: ImmutableMetricData{resource=Resource{schemaUrl=null, attributes={namespace="dev", version="dev", visibility.libversion="4.0.0", visibility.mode="dev", visibility.service="dev"}}, instrumentationScopeInfo=InstrumentationScopeInfo{name=io.opentelemetry.runtime-metrics, version=1.24.0-alpha, schemaUrl=null, attributes={}}, name=process.runtime.jvm.gc.duration, description=Duration of JVM garbage collection actions, unit=ms, type=HISTOGRAM, data=ImmutableHistogramData{aggregationTemporality=CUMULATIVE, points=[ImmutableHistogramPointData{getStartEpochNanos=1681906955619000000, getEpochNanos=1681907335625000000, getAttributes={action="end of minor GC", gc="PS Scavenge"}, getSum=34.0, getCount=1, hasMin=true, getMin=34.0, hasMax=true, getMax=34.0, getBoundaries=[5.0, 10.0, 25.0, 50.0, 75.0, 100.0, 250.0, 500.0, 750.0, 1000.0, 2500.0, 5000.0, 7500.0, 10000.0, 25000.0, 50000.0, 75000.0, 100000.0, 250000.0, 500000.0, 750000.0, 1000000.0], getCounts=[0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0], getExemplars=[]}]}}
```
Any suggestion would be welcomed 🙏

**Comment by trask** [comment]
can you enable GC logging `-verbose:gc -XX:+PrintGCDetails` and confirm that you are seeing other kinds of GC events there?

**Comment by yorikya** [comment]
Sure, will do it soon, thank you

**Comment by yorikya** [comment]
Yes, I am seeing GC events:

```
[GC (Allocation Failure) [PSYoungGen: 65536K->8369K(76288K)] 65536K->8385K(251392K), 0.0063642 secs] [Times: user=0.02 sys=0.01, real=0.00 secs] 
[GC (Allocation Failure) [PSYoungGen: 73905K->7517K(76288K)] 73921K->7549K(251392K), 0.0087471 secs] [Times: user=0.05 sys=0.00, real=0.01 secs] 
[GC (Allocation Failure) [PSYoungGen: 73053K->6953K(76288K)] 73085K->6993K(251392K), 0.0055033 secs] [Times: user=0.04 sys=0.00, real=0.01 secs] 
[GC (Allocation Failure) [PSYoungGen: 72489K->7518K(141824K)] 72529K->7566K(316928K), 0.0078794 secs] [Times: user=0.06 sys=0.01, real=0.01 secs] 
[GC (Metadata GC Threshold) [PSYoungGen: 15666K->7568K(141824K)] 15714K->7624K(316928K), 0.0090825 secs] [Times: user=0.06 sys=0.01, real=0.01 secs] 
[Full GC (Metadata GC Threshold) [PSYoungGen: 7568K->0K(141824K)] [ParOldGen: 56K->7364K(86016K)] 7624K->7364K(227840K), [Metaspace: 19223K->19223K(1067008K)], 0.0370681 secs] [Times: user=0.21 sys=0.01, real=0.03 secs] 
[GC (Allocation Failure) [PSYoungGen: 131072K->1461K(232960K)] 138436K->8833K(318976K), 0.0034432 secs] [Times: user=0.02 sys=0.00, real=0.01 secs] 

...

[GC (Allocation Failure) [PSYoungGen: 252928K->11257K(396800K)] 269913K->29284K(525312K), 0.0105141 secs] [Times: user=0.04 sys=0.01, real=0.01 secs] 
```

This is very strange I don't change/add JVM options in my service, just upgrade the runtime-metrics lib and I don't see the `marksweep` anymore. 

Just for note: My service is written in Clojure and I am using Java OT SDK in order to submit the metric and traces, This service runs with the runtime-metrics for almost 2 years. 

runtime-metrics "1.19.2-alpha"
![Screenshot 2023-04-20 at 14 17 03](https://user-images.githubusercontent.com/46996631/233350817-fc4c7ad1-ea21-4e1d-a83e-be3154c02f7e.png)



runtime-metrics "1.24.0-alpha"
![Screenshot 2023-04-20 at 14 18 24](https://user-images.githubusercontent.com/46996631/233350676-48b1eb39-25ab-4f6e-91bd-c733c459d58a.png)

**Comment by PeterF778** [comment]
If you put a lot of load on your application, it will eventually have to run marksweep gc. I'd also try to use `jconsole` and trigger Full GC manually.

**Comment by laurit** [comment]
@yorikya Do I understand correctly that even when you see `Full GC` in gc log you don't see anything besides `PS Scavenge` in collector?

**Comment by yorikya** [comment]
@laurit Yes you are correct 👍 

@PeterF778 I will try to decrease the `xmx/xms` and then will try to keep the `marksweep` gc, also I'll try the `jconsole`, thanks 🙏
